include/NovelRT/Maths/GeoVector2F.h

```diff
@@ -21,157 +24,789 @@ namespace NovelRT::Maths
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the entrywise sum of this and another scalar and applies it to
+         * this instance.
+         *
+         * @details
+         * An example of summing up a vector and a scalar: \f[
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      +
+         *      6
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      +
+         *      \begin{pmatrix}
+         *      6\\
+         *      6
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      8 + 6 \\
+         *      6 + 6
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      14\\
+         *      12
+         *      \end{pmatrix}
+         * \f]
+         * @param other The other scalar that gets added up with each component of this vector.
+         * @return This GeoVecor2F who's components consist of the entrywise sum of this and another scalar.
+         */
         inline GeoVector2F operator+=(float other) noexcept
         {
             *reinterpret_cast<glm::vec2*>(this) = *reinterpret_cast<const glm::vec2*>(this) + other;
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the entrywise difference of this and another scalar and applies it
+         * to this instance.
+         *
+         * @details
+         * An example of subtracting a scalar from a vector: \f[
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      -
+         *      3
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      8\\
+         *      6
+         *      \end{pmatrix}
+         *      -
+         *      \begin{pmatrix}
+         *      3\\
+         *      3
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      8 - 3 \\
+         *      6 - 3
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      5\\
+         *      3
+         *      \end{pmatrix}
+         * \f]
+         * @param other The other scalar that gets subtracted from each component of this vector.
+         * @return This GeoVector2F who's components consist of the entrywise difference of this and another scalar.
+         */
         inline GeoVector2F operator-=(float other) noexcept
         {
             *reinterpret_cast<glm::vec2*>(this) = *reinterpret_cast<const glm::vec2*>(this) - other;
             return *this;
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the product of this and another scalar.
```

Comment:
construct on operator. Also this is in-place memory overwriting as it returns `T&` to `this`.